pkg/devfile/parser/writer.go

```diff
@@ -30,3 +38,58 @@ func (d *DevfileObj) WriteYamlDevfile() error {
 	klog.V(2).Infof("devfile yaml created at: '%s'", OutputDevfileYamlPath)
 	return nil
 }
+
+func restoreK8sCompURI(devObj *DevfileObj) error {
+	getKubeCompOptions := common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: v1.KubernetesComponentType,
+		},
+	}
+	getOpenshiftCompOptions := common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: v1.OpenshiftComponentType,
+		},
+	}
+	kubeComponents, err := devObj.Data.GetComponents(getKubeCompOptions)
+	if err != nil {
+		return err
+	}
+	openshiftComponents, err := devObj.Data.GetComponents(getOpenshiftCompOptions)
+	if err != nil {
+		return err
+	}
+	for _, kubeComp := range kubeComponents {
+		var keyNotFoundErr = &apiAttributes.KeyNotFoundError{Key: K8sLikeComponentOriginalURIKey}
+		uri := kubeComp.Attributes.GetString(K8sLikeComponentOriginalURIKey, &err)
+		if err != nil && err.Error() != keyNotFoundErr.Error() {
+			return err
+		}
```

Comment:
you can also do it this way:

```suggestion
		if err != nil {
		        if _, ok := err.(*attributes.KeyNotFoundError); !ok {
			        return err
		        }
	        }
```

Reply:
updated